ov/scope.go

```diff
@@ -0,0 +1,206 @@
+package ov
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type Scope struct {
+	Description   utils.Nstring    `json:"description,omitempty"`   // "description": "Test from script",
+	Modified      string           `json:"modified,omitempty"`      // "modified": "2018-12-13T10:24:25.267Z",
+	Name          string           `json:"name,omitempty"`          // "name": "updated-SD3",
+	State         string           `json:"state,omitempty"`         // "state": "null",
+	Status        string           `json:"status,omitempty"`        // "status": "null",
+	Type          string           `json:"type,omitempty"`          // "type": "scopesV3",
+	URI           utils.Nstring    `json:"uri,omitempty"`           // "uri": "/rest/scopes/7f658031-c942-4336-be7a-67957cf20ba2"
+	ExtAttributes *ExtraAttributes `json:"extAttributes,omitempty"` //{}
+	ApplianceId   string           `json:"applianceId,omitempty"`   // "category": "scopes",
+	Category      string           `json:"category,omitempty"`      // "category": "scopes",
+	Created       string           `json:"created,omitempty"`       // "created": "2018-12-13T10:05:35.745Z",
+	ETAG          string           `json:"eTag,omitempty"`          // "eTag": "\"2018-12-13T10:24:25.267Z/2018-12-13T10:24:25.267Z\"",
+	OldUri        utils.Nstring    `json:"oldUri,omitempty"`        //"oldUri": "null",
+	ScopesUri     utils.Nstring    `json:"scopesUri,omitempty"`     //"scopesUri": "/rest/scopes/resources/rest/scopes/7f658031-c942-4336-be7a-67957cf20ba2"
+}
+
+type ExtraAttributes struct {
+	Type string `json:"type,omitempty"`
+}
+
+type ScopeList struct {
+	Total       int           `json:"total,omitempty"`       // "total": 1,
+	Count       int           `json:"count,omitempty"`       // "count": 1,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	URI         utils.Nstring `json:"uri,omitempty"`         // "uri": "/rest/scopes"
+	Members     []Scope       `json:"members,omitempty"`     // "members":[]
+}
+
+func (c *OVClient) GetScopeByName(name string) (Scope, error) {
+	var (
+		scp Scope
+	)
+	scps, err := c.GetScopes(fmt.Sprintf("name matches '%s'", name), "name:asc")
+	if scps.Total > 0 {
+		return scps.Members[0], err
+	} else {
+		return scp, err
+	}
+}
+
+func (c *OVClient) GetScopes(query string, sort string) (ScopeList, error) {
```

Comment:
query param name should be more specific here..

Reply:
in the API reference doc parameter by name query is there .

Reply:
you can ignore then